viewing_party/party.py

```diff
@@ -1,23 +1,182 @@
+    # Return None if list of "watched" is empty
+    if not user_data["watched"]:
+        return None
+
+    # Count watched genres
+    genre_count = {}
+    for movie in user_data["watched"]:
+        if movie["genre"] not in genre_count:
+            genre_count[movie["genre"]] = 1
+        else:
+            genre_count[movie["genre"]] += 1
+
+    # Determine most watched genre
+    max_num = 0
+    most_watched_genre = None
+    for genre in genre_count:
+        if genre_count[genre] > max_num:
+            max_num = genre_count[genre]
+            most_watched_genre = genre
+
+    return most_watched_genre
+
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+# Add helper function to get movie titles
+def get_titles(list_of_dict):
+    return {item["title"] for item in list_of_dict}
+
+# Get all movies unique to user
+def get_unique_watched(user_data):
+
+    # Put titles that friends watched in a set
+    friends_movies = set()
+    for friend in user_data["friends"]:
+        friends_movies.update(get_titles(friend["watched"]))
+
+    # Create list of movies that only user watched
+    unique_movies = [
+        movie for movie in user_data["watched"] 
+        if movie["title"] not in friends_movies
+    ]
+
+    return unique_movies
+
+# Get movies unique to friends that user hasn't watched
+def get_friends_unique_watched(user_data):
+
+    # Put movie titles user has watched in a set
+    user_movie_titles = get_titles(user_data["watched"])
+
+    # Create a list for movies only friends watched
+    unique_friends_movies = []
+
+    # Put movies friends watched to a set
+    friends_movie_titles = set()
+   
+    # Iterate through movies, if movie title is not in a set 
+    # unique to a user and if movie title is not in friends movies
+    # add it to a set of titles unique to friends, so that we don't add
+    # duplicates to unique friend's movies. Append movies 
+    # to a list of unique movies 
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            title = movie["title"]
+            if title not in user_movie_titles:
+                if title not in friends_movie_titles:
+                    friends_movie_titles.add(title)
+                    unique_friends_movies.append(movie)
+
+    return unique_friends_movies
+
+
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+# Get friends' recommendations for user
+def get_available_recs(user_data):
+
+    # Get unique movies friends watched, but user hasn't
+    unique_movies = get_friends_unique_watched(user_data)
+
+    recommended_movies = []
+    # Append a movie to a list of recommended movies
+    # if a movie is hosted by a subscription user has acces to
+    for movie in unique_movies:
+        host = movie["host"]

```

Comment:
I'm just calling this out because it ends up being a little inconsistent from function to function. Sometimes you store data from the dictionary in a new variable, other times you don't. As you get more comfortable with programming, I would lean toward directly accessing the data, but more importantly, keep things consistent across your program!